core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -77,6 +77,16 @@ private[spark] class AppClient(
     private val registrationRetryThread =
       ThreadUtils.newDaemonSingleThreadScheduledExecutor("appclient-registration-retry-thread")
 
+    private val receiveAndReplyMaxPoolSize = conf.getInt(
```

Comment:
I was hoping @zsxwing could chime in and help me understand the use cases for this.  I just arbitrarily chose 3 because it didn't seem to me like these messages would come in a flurry and need many threads, but I don't know for sure.  Maybe a configurable max is also not necessary.